*** src/interpolate/test/ppm_test_interp_m2p.f	2010-11-12 17:25:10.000000000 +0100
--- src/interpolate/test/ppm_test_interp_m2p.f.new	2011-07-13 22:32:21.090448794 +0200
***************
*** 36,44 ****
      use ppm_module_init
      use ppm_module_finalize
      use ppm_module_map_part_global
!     use ppm_module_map_part_push
!     use ppm_module_map_part_pop
!     use ppm_module_map_part_send
  
      implicit none
  
--- 36,42 ----
      use ppm_module_init
      use ppm_module_finalize
      use ppm_module_map_part_global
!     use ppm_module_map_part
  
      implicit none
  
